### PR TITLE
Add support for ratio driven heights + tooltip fades

### PR DESCRIPTION
**Pull request by KrisSiegel**
This PR adds support for ratio driven heights to avoid the bleeding issue when BoPattern is contained within a non-explicit height (this seems like a typical solution though I'd like to do another PR later with a "golden ratio" where it defaults to using this when it detects that the parent element will keep expanding).

Also added in some tooltip fading. This is the last bug I needed to fix to get this properly integrated into BoData console.

@immuta/developers

### CODE REVIEW

File: dist/bopattern.js [hunk 1]
```diff
@@ -766,11 +772,9 @@ BoPattern.extend(function(internal) {
...
             tooltip.style.left = (((e.width / 2) + e.absX) - (tooltip.getBoundingClientRect().width / 2)) + "px";
             tooltip.style.top = (e.absY - e.height) + "px";
-            tooltip.style.display = "block";
+            tooltip.className = "polTooltip in";
             timeout = setTimeout(function() {
-                tooltip.style.display = "none";
-                tooltip.style.left = 0;
-                tooltip.style.top = 0;
+                tooltip.className = "polTooltip out";
             }, 3500);
```
Comment (ericjperry) [inline dist/bopattern.js]: Does the tooltip fade out after 3.5 seconds even if you continue to hover in the cell?

File: dist/bopattern.js [hunk 2]
```diff
@@ -22,6 +22,9 @@ var BoPattern = (function() {
             data: {},
             eventHandlers: {},
             config: msngr.merge({
+                screen: {
+                    radio: undefined
```
Comment (jeffmhastings) [inline dist/bopattern.js]: radio -> ratio

File: src/main.js [hunk 3]
```diff
@@ -22,6 +22,9 @@ var BoPattern = (function() {
             data: {},
             eventHandlers: {},
             config: msngr.merge({
+                screen: {
+                    radio: undefined
```
Comment (jeffmhastings) [inline src/main.js]: radio -> ratio

**Comment by jeffmhastings** [comment]
:+1:

File: dist/bopattern.js (hunk 1 above)
Comment (KrisSiegel) [inline dist/bopattern.js]: Yeah it does unfortunately. Implementing a hoverout event was kind of a pain so I left it for some future 2.0 release :P

File: src/main.js (hunk 3 above)
Comment (KrisSiegel) [inline src/main.js]: Do'h. Fixed

File: dist/bopattern.js (hunk 1 above)
Comment (ericjperry) [inline dist/bopattern.js]: Understood.

**Comment by ericjperry** [comment]
:+1:

**Pull request merged**